src/main/scala/mesosphere/marathon/core/pod/PodDefinition.scala

```diff
@@ -107,7 +107,11 @@ case class PodDefinition(
   }
 
   def container(name: String): Option[MesosContainer] = containers.find(_.name == name)
-  def container(taskId: Task.Id): Option[MesosContainer] = taskId.containerName.flatMap(container(_))
+  def container(taskId: Task.Id): Option[MesosContainer] = taskId match {
+    case Task.EphemeralOrReservedTaskId(_, Some(containerName)) => container(containerName)
```

Comment:
Here too: what do you think of having `Task.Id.containerName` and using it here?

Reply:
✅